.github/workflows/worker-deploy.yml

```diff
@@ -31,23 +34,26 @@ jobs:
           # Update the wrangler.toml file
           sed -i "s/^name = .*/name = \"$new_name\"/" wrangler.toml
           echo "Updated wrangler.toml name to: $new_name"
+
       - name: Deploy with Wrangler
         id: wrangler_deploy
         uses: cloudflare/wrangler-action@v3
         with:
-          wranglerVersion: "3.79.0"
+          wranglerVersion: "3.87.0"
           apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
           accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
           secrets: |
             SUPABASE_URL
             SUPABASE_KEY
             CLOUDFLARE_ACCOUNT_ID
             CLOUDFLARE_API_TOKEN
+            ${{ secrets.KERNEL_PUBLIC_KEY && secrets.KERNEL_PUBLIC_KEY != '' && 'KERNEL_PUBLIC_KEY' || '' }}
```

Comment:
You are missing the environment switch to development for this to properly work.